src/Orleans/Runtime/AsynchAgent.cs

```diff
@@ -153,6 +157,23 @@ public void Abort(object stateInfo)
                 t.Abort(stateInfo);
         }
 
+        public void Join(TimeSpan timeout)
```

Comment:
Added the Join method, although there are currently no usages of it. We may consider using it in the future.